From 7fb4961d53d878232744de68fa62862136ec0482 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Sun, 27 Apr 2025 15:24:00 +0200
Subject: [PATCH] Plugwise: include appliance name in sensor name

Multiple devices in an Anna/Adam system can provide the same measurement
, e.g. 'temperature'. Add a namespace to be able to distinguish between
them. The name set in the Plugwise UI will be used.

This changes the default name from e.g. "temperature" to "Jip Zithoek
temperature".

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 34 +++++++++++++++++-----------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index 6dfc8148c..0eec88190 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -511,7 +511,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float temperature = (float)atof(tmpstr.c_str());
-					SendTempSensor(applianceIDhex, 255, temperature, sname);
+					SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "illuminance")
@@ -520,7 +520,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float illuminance = (float)atof(tmpstr.c_str());
-					SendLuxSensor(applianceIDhex, 1, 255, illuminance, sname);
+					SendLuxSensor(applianceIDhex, 1, 255, illuminance, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "thermostat")
@@ -529,7 +529,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float temperature = (float)atof(tmpstr.c_str());
-					SendSetPointSensor(0, 0, 0, sAnnaThermostat, 1, 255, temperature, sname);
+					SendSetPointSensor(0, 0, 0, sAnnaThermostat, 1, 255, temperature, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "intended_boiler_temperature")
@@ -538,7 +538,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float temperature = (float)atof(tmpstr.c_str());
-					SendTempSensor(applianceIDhex, 255, temperature, sname);
+					SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "return_water_temperature")
@@ -547,7 +547,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float temperature = (float)atof(tmpstr.c_str());
-					SendTempSensor(applianceIDhex, 255, temperature, sname);
+					SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "boiler_temperature")
@@ -556,7 +556,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float temperature = (float)atof(tmpstr.c_str());
-					SendTempSensor(applianceIDhex, 255, temperature, sname);
+					SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "maximum_boiler_temperature")
@@ -565,7 +565,7 @@ void CAnnaThermostat::GetMeterDetails()
 				if (!tmpstr.empty())
 				{
 					float temperature = (float)atof(tmpstr.c_str());
-					SendTempSensor(applianceIDhex, 255, temperature, sname);
+					SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
 				}
 			}
 			else if (sname == "boiler_state")
@@ -575,11 +575,11 @@ void CAnnaThermostat::GetMeterDetails()
 				{
 					if (strcmp(tmpstr.c_str(), "on") == 0)
 					{
-						SendSwitch(applianceIDhex, 1, 255, true, 0, sname, m_Name);
+						SendSwitch(applianceIDhex, 1, 255, true, 0, ApplianceName + " " + sname, m_Name);
 					}
 					else
 					{
-						SendSwitch(applianceIDhex, 1, 255, false, 0, sname, m_Name);
+						SendSwitch(applianceIDhex, 1, 255, false, 0, ApplianceName + " " + sname, m_Name);
 					}
 				}
 			}
@@ -591,11 +591,11 @@ void CAnnaThermostat::GetMeterDetails()
 					if (strcmp(tmpstr.c_str(), "on") == 0)
 					{
 
-						SendSwitch(applianceIDhex, 1, 255, true, 0, sname, m_Name);
+						SendSwitch(applianceIDhex, 1, 255, true, 0, ApplianceName + " " + sname, m_Name);
 					}
 					else
 					{
-						SendSwitch(applianceIDhex, 1, 255, false, 0, sname, m_Name);
+						SendSwitch(applianceIDhex, 1, 255, false, 0, ApplianceName + " " + sname, m_Name);
 					}
 				}
 			}
@@ -624,11 +624,11 @@ void CAnnaThermostat::GetMeterDetails()
 				{
 					if (strcmp(tmpstr.c_str(), "on") == 0)
 					{
-						SendSwitch(applianceIDhex, 1, 255, true, 0, sname, m_Name);
+						SendSwitch(applianceIDhex, 1, 255, true, 0, ApplianceName + " " + sname, m_Name);
 					}
 					else
 					{
-						SendSwitch(applianceIDhex, 1, 255, false, 0, sname, m_Name);
+						SendSwitch(applianceIDhex, 1, 255, false, 0, ApplianceName + " " + sname, m_Name);
 					}
 				}
 			}
@@ -678,7 +678,7 @@ void CAnnaThermostat::GetMeterDetails()
                 if (!tmpstr.empty())
                 {
                     float level = (float)atof(tmpstr.c_str()) * 100.F;
-                    SendPercentageSensor(applianceIDhex, 1, 255, level, sname);
+                    SendPercentageSensor(applianceIDhex, 1, 255, level, ApplianceName + " " + sname);;
                 }
             }
             else if (sname == "central_heater_water_pressure")
@@ -687,7 +687,7 @@ void CAnnaThermostat::GetMeterDetails()
                 if (!tmpstr.empty())
                 {
                     float pressure = (float)atof(tmpstr.c_str());
-                    SendPressureSensor(applianceIDhex, 1, 255, pressure, sname);
+                    SendPressureSensor(applianceIDhex, 1, 255, pressure, ApplianceName + " " + sname);;
                 }
             }
             else if (sname == "signal_strength")
@@ -741,7 +741,7 @@ void CAnnaThermostat::GetMeterDetails()
                 if (!tmpstr.empty())
                 {
                     float flow = (float)atof(tmpstr.c_str());
-                    SendWaterflowSensor(applianceIDhex, 1, 255, flow, sname);
+                    SendWaterflowSensor(applianceIDhex, 1, 255, flow, ApplianceName + " " + sname);;
                 }
             }
             else if (sname == "domestic_hot_water_temperature")
@@ -750,7 +750,7 @@ void CAnnaThermostat::GetMeterDetails()
                 if (!tmpstr.empty())
                 {
                     float temperature = (float)atof(tmpstr.c_str());
-                    SendTempSensor(applianceIDhex, 255, temperature, sname);
+                    SendTempSensor(applianceIDhex, 255, temperature, ApplianceName + " " + sname);;
                 }
             }
         }
-- 
2.47.3

